### PR TITLE
Implement initial --webgpu-mode flag

### LINKED ISSUE #2166: Create high level recipe for WebGPU in optimizer
This should at first just do a dead code removal pass.

### PR DESCRIPTION
**Pull request by zoddicus**
Fixes #2166

### CODE REVIEW

**Comment by zoddicus** [comment]
@s-perron, can you confirm that these are sensible flags to be setting for dead code elimation? I suspect agressive might already cover the other flags.

File: source/opt/optimizer.cpp [hunk 1]
```diff
@@ -216,6 +216,12 @@ Optimizer& Optimizer::RegisterSizePasses() {
       .RegisterPass(CreateAggressiveDCEPass());
 }
 
+Optimizer& Optimizer::RegisterWebGPUPasses() {
+  return RegisterPass(CreateEliminateDeadFunctionsPass())
+      .RegisterPass(CreateAggressiveDCEPass())
+      .RegisterPass(CreateEliminateDeadConstantPass());
```
Comment (s-perron) [inline source/opt/optimizer.cpp]: If you really just want removal of dead code, and not something that will look to simplify obvious stuff, you should do AggressiveDCEPass and DeadBranchElimination.  The other two that you have are redundant when you have ADCE.

File: tools/opt/opt.cpp [hunk 2]
```diff
@@ -359,6 +359,18 @@ Options (in lexicographical order):
...
+  --webgpu-mode
+               Turns on the prescribed passes for WebGPU and sets the target
+               environmet to webgpu0. Other passes may be turned on via
+               additional flags, but such combinations are not tested.
+               Using --target-env with this flag is not allowed.
+               This flag is the equivalent of setting the following flags:
+                 --target-env=webgpu0
+                 --eliminate-dead-functions
+                 --eliminate-dead-code-aggressive
+                 --eliminate-dead-const
```
Comment (s-perron) [inline tools/opt/opt.cpp]: There is a way of getting the options for the optimization passes printed without hard coding the order here.  See how -O and -Os work.

File: source/opt/optimizer.cpp (hunk 1 above)
Comment (zoddicus) [reply source/opt/optimizer.cpp]: Done

File: tools/opt/opt.cpp (hunk 2 above)
Comment (zoddicus) [reply tools/opt/opt.cpp]: Done

**Pull request merged**